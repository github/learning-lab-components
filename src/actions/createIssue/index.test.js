const createIssue = require('.')
const mockContext = require('../../tests/mockContext')

describe('createIssue', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      issues: {
        create: jest.fn(() => Promise.resolve({ data: { number: 1 } })),
        createComment: jest.fn()
      }
    })
  })

  it('opens an issue', async () => {
    await createIssue(context, { title: 'My issue', body: 'a-body.md', data: { foo: true } })
    expect(context.github.issues.create).toHaveBeenCalled()
    expect(context.github.issues.create.mock.calls).toMatchSnapshot()

    // Ensures that data was passed
    expect(context.fromFile.mock.calls).toMatchSnapshot()
  })

  it('opens an issue with comments', async () => {
    await createIssue(context, { title: 'My issue', body: 'a-body.md', comments: ['a-comment.md', 'another-comment.md'], data: { foo: true } })
    expect(context.github.issues.create).toHaveBeenCalled()
    expect(context.github.issues.createComment).toHaveBeenCalledTimes(2)
    expect(context.github.issues.createComment.mock.calls).toMatchSnapshot()

    // Ensures that data was passed
    expect(context.fromFile.mock.calls).toMatchSnapshot()
  })
})
