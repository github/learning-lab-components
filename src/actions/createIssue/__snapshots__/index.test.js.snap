// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createIssue opens an issue 1`] = `
Array [
  Array [
    Object {
      "body": "a-body.md",
      "owner": "JasonEtco",
      "repo": "example",
      "title": "My issue",
    },
  ],
]
`;

exports[`createIssue opens an issue 2`] = `
Array [
  Array [
    "a-body.md",
    Object {
      "foo": true,
    },
  ],
]
`;

exports[`createIssue opens an issue with comments 1`] = `
Array [
  Array [
    Object {
      "body": "a-comment.md",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
  Array [
    Object {
      "body": "another-comment.md",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`createIssue opens an issue with comments 2`] = `
Array [
  Array [
    "a-body.md",
    Object {
      "foo": true,
    },
  ],
  Array [
    "a-comment.md",
    Object {
      "foo": true,
    },
  ],
  Array [
    "another-comment.md",
    Object {
      "foo": true,
    },
  ],
]
`;
